sycl/include/sycl/vector.hpp

```diff
@@ -111,24 +97,807 @@ template <typename T> class GetOp {
 //
 // must go throw `v.x()` returning a swizzle, then its `operator==` returning
 // vec<int, 1> and we want that code to compile.
-template <typename Vec, typename T, int N, typename = void>
+template <typename Self, typename T, int N, typename = void>
 struct ScalarConversionOperatorMixIn {};
 
-template <typename Vec, typename T, int N>
-struct ScalarConversionOperatorMixIn<Vec, T, N, std::enable_if_t<N == 1>> {
-  operator T() const { return (*static_cast<const Vec *>(this))[0]; }
+template <typename Self, typename T, int N>
+struct ScalarConversionOperatorMixIn<Self, T, N, std::enable_if_t<N == 1>> {
+  operator T() const { return (*static_cast<const Self *>(this))[0]; }
+};
+
+// Swizzle's `operator vec<DataT, NumElements>() const` has `NumElements > 1`
+// constraint.
+template <typename Self, typename T, int... Indexes>
+struct SwizzleToVectorConversionOperatorMixIn {
+  operator vec<T, sizeof...(Indexes)>() const {
+    int idx = 0;
+    return {(*static_cast<const Self *>(this))[Indexes, idx++]...};
```

Comment:
This is deprecated in C++20 and incorrect in C++23. We want our headers to work if the user uses a newer C++ version, correct?
```suggestion
    return {(*static_cast<const Self *>(this))[(Indexes, idx++)]...};
```

Reply:
Hi @rolandschulz , I haven't looked into future yet... Are you saying that what the PR currently has will stop working, but your suggested change would address that, or is something else required?